lib/matplotlib/cbook.py

```diff
@@ -2344,6 +2344,26 @@ def _picklable_class_constructor(mixin_class, fmt, attr_name, base_class):
     return cls.__new__(cls)
 
 
+def _is_torch_array(x):
+    """Check if 'x' is a PyTorch Tensor."""
+    try:
+        # we're intentionally not attempting to import torch. If somebody
+        # has created a torch array, torch should already be in sys.modules
+        return isinstance(x, sys.modules['torch'].Tensor)
+    except Exception:
```

Comment:
I would add a note here about why we're using raw `Exception`, because someone will question this in the future!

Reply:
@jakevdp That's true. How about the following note?

```py
except Exception:
    # we're catching all exceptions intentionally during isinstance check to 
    # prevent potential unknown exceptions and avoid `matplotlib` crashes.
    return False
```

Reply:
I'd say something more specific, like `we're attempting to access attributes on imported modules which may have arbitrary user code, so we deliberately catch all exceptions`.